components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/query/QueryFactory.java

```diff
@@ -1205,7 +1235,9 @@ private static StaticOutputElement createStaticOutputElement(DataService dataSer
 		/* workaround for different character case issues in column names,
 		 * constant values will be as it is */
 		if (!DBSFields.VALUE.equals(paramType)) {
-		    param = param.toLowerCase();
+			if (!isCaseSensitive) {
```

Comment:
Merge if statements to have a single one with the above one. Fix in other places too.